### PR TITLE
[Add] Add command group `microsoft.datafactory/factories`

### PR DESCRIPTION
**Pull request by wu5bocheng**
|command | status|
| :-------------------: | :--------: |
| datafactory list   | done |
| datafactory show   | done |
| datafactory delete | done |

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: data-model-for-portal/Microsoft.DataFactory/datafactory.json [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                        "parameters": [
+                            "[--resource-group]",
+                            "[--subscription]"
+                        ]
+                    }
+                ]
+            },
+            "examples": [
+                {
+                    "description": "Factories_ListByResourceGroup",
```
Comment (jsntcy) [inline data-model-for-portal/Microsoft.DataFactory/datafactory.json]: Description can be optimized

File: data-model-for-portal/Microsoft.DataFactory/datafactory.json [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                            "[--if-none-match]",
+                            "[--resource-group]",
+                            "[--subscription]"
+                        ]
+                    }
+                ]
+            },
+            "examples": [
+                {
+                    "description": "Factories_Get",
```
Comment (jsntcy) [inline data-model-for-portal/Microsoft.DataFactory/datafactory.json]: Description can be optimized

**Comment by wu5bocheng** [comment]
Modified, can be re-examined and merged

File: data-model-for-portal/Microsoft.DataFactory/datafactory.json [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            "path": "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}",
+            "confirmation": false,
+            "help": {
+                "learnMore": {
+                    "url": "https://docs.microsoft.com/cli/azure/datafactory#az-datafactory-show"
+                },
+                "parameterSets": [
+                    {
+                        "parameters": [
+                            "[--factory-name]",
```
Comment (jsntcy) [inline data-model-for-portal/Microsoft.DataFactory/datafactory.json]: Use `--name` to keep consistent with example below.
Same for `az datafactory delete`

File: data-model-for-portal/Microsoft.DataFactory/datafactory.json [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            "path": "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}",
+            "confirmation": true,
+            "help": {
+                "learnMore": {
+                    "url": "https://docs.microsoft.com/cli/azure/datafactory#az-datafactory-delete"
+                },
+                "parameterSets": [
+                    {
+                        "parameters": [
+                            "[--factory-name]",
```
Comment (jsntcy) [inline data-model-for-portal/Microsoft.DataFactory/datafactory.json]: consistent issue

**Review by jsntcy — APPROVED** [review]

**Pull request merged**